### PR TITLE
[Fix] Merge resize and sr_resize.

### PR DESCRIPTION
**Pull request by Yshuo-Li**

### CODE REVIEW

File: mmedit/datasets/pipelines/augmentation.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ class Resize:
         keys (list[str]): The images to be resized.
         scale (float | Tuple[int]): If scale is Tuple(int), target spatial
             size (h, w). Otherwise, target spatial size is scaled by input
-            size. If any of scale is -1, we will rescale short edge.
+            size.
```
Comment (innerlee) [inline mmedit/datasets/pipelines/augmentation.py:36]: Reason for the deletion?

File: mmedit/datasets/pipelines/augmentation.py [hunk 2]
```diff
@@ -50,6 +50,12 @@ class Resize:
         interpolation (str): Algorithm used for interpolation:
             "nearest" | "bilinear" | "bicubic" | "area" | "lanczos".
             Default: "bilinear".
+        backend (str | None): The image resize backend type. Options are `cv2`,
+            `pillow`, `None`. If backend is None, the global imread_backend
+            specified by ``mmcv.use_backend()`` will be used.
+            Default: "pillow".
```
Comment (innerlee) [inline mmedit/datasets/pipelines/augmentation.py:56]: Why choose pillow?

File: tests/test_augmentation.py [hunk 3]
```diff
@@ -673,9 +680,15 @@ def mirror_sequence(self):
...
-            f"(keys=['lq', 'gt'])")
+            "(keys=['lq', 'gt'])")
 
         # each key should contain a list of nparray
         with pytest.raises(TypeError):
             results = dict(lq=0, gt=gts)
             mirror_sequence(results)
+
+
+if __name__ == '__main__':
```
Comment (innerlee) [inline tests/test_augmentation.py:691]: No main in test files

File: tests/test_augmentation.py [hunk 4]
```diff
@@ -461,14 +461,21 @@ def test_resize(self):
...
-        assert results['gt_img'].shape == (128, 128, 1)
+        assert results['gt_img'].shape == (256, 256)
+        assert results['lq_img'].shape == (128, 128, 1)
+        assert results['alpha'].shape == (240, 320)
+        assert results['beta'].shape == (128, 128, 1)
 
         name_ = str(resize_keep_ratio)
         assert name_ == resize_keep_ratio.__class__.__name__ + (
-            f"(keys={['gt_img']}, scale=(128, 128), "
+            f"(keys={['gt_img']}, output_keys={['gt_img']}, "
```
Comment (innerlee) [inline tests/test_augmentation.py:477]: What the point of `{['gt_img']}`

File: tests/test_augmentation.py [hunk 5]
```diff
@@ -461,14 +461,21 @@ def test_resize(self):
         assert results['gt_img'].shape[:2] == (128, 128)
 
         # test input with shape (256, 256)
-        results = dict(gt_img=self.results['img'][..., 0].copy())
-        resize = Resize(['gt_img'], scale=(128, 128), keep_ratio=False)
+        results = dict(gt_img=self.results['img'][..., 0].copy(), alpha=alpha)
+        resize = Resize(['gt_img', 'alpha'],
+                        scale=(128, 128),
```
Comment (innerlee) [inline tests/test_augmentation.py:466]: I thought what you want is a relative-scaling? Why not test it?

File: mmedit/datasets/pipelines/augmentation.py (hunk 2 above)
Comment (Yshuo-Li) [reply mmedit/datasets/pipelines/augmentation.py:56]: Sorry, it is copied from another function, default: 'cv2'

File: tests/test_augmentation.py (hunk 5 above)
Comment (Yshuo-Li) [reply tests/test_augmentation.py:466]: mmcv.imrescale supports scale (float | Tuple[int]), so we don't need relative-scaling.

Comment (innerlee) [reply tests/test_augmentation.py:466]: Okay.

**Review by innerlee — APPROVED** [review]

File: mmedit/datasets/pipelines/augmentation.py [hunk 6]
```diff
@@ -50,6 +50,12 @@ class Resize:
         interpolation (str): Algorithm used for interpolation:
             "nearest" | "bilinear" | "bicubic" | "area" | "lanczos".
             Default: "bilinear".
+        backend (str | None): The image resize backend type. Options are `cv2`,
+            `pillow`, `None`. If backend is None, the global imread_backend
+            specified by ``mmcv.use_backend()`` will be used.
+            Default: "cv2".
```
Comment (innerlee) [inline mmedit/datasets/pipelines/augmentation.py:56]: Again, the default backend should be None. Otherwise you will change behavior of old codes

Comment (Yshuo-Li) [reply mmedit/datasets/pipelines/augmentation.py:56]: OK

Comment (Yshuo-Li) [reply mmedit/datasets/pipelines/augmentation.py]: In fact, if backend=None, cv2 will be used in mmcv.resize

**Pull request merged**